Controller/ThemeController.php

```diff
@@ -92,4 +91,16 @@ public function switchAction(Request $request)
 
         return $response;
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function extractUrl(Request $request)
+    {
+        $url = $request->headers->get('Referer');
+
+        return !empty($url) ? $url : '/';
```

Comment:
I guess `empty()` would also return true on `0` but wouldn't really make a lot of sense to have as a referer, so I guess it would be ok to then to default to `/` ..

however I would flip the condition to just be `return empty($url) ? '/' : $url;` to avoid the double negative

Reply:
flipped